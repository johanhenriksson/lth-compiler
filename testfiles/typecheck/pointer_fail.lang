void func() {
    int a;
    bool* bool_ptr = &a; // error;
}

int main() {
    bool a;
    bool* bool_ptr = &a;
    return *bool_ptr; // error
}
