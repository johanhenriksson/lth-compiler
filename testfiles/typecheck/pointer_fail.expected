Error at line 3,5: cannot assign 'bool_ptr' of type bool*, got int*
Error at line 9,5: expected return type int, got bool
